clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -447,28 +447,36 @@ object GarbageCollector {
     val numAddressPartitions = hc.getInt(LAKEFS_CONF_GC_NUM_ADDRESS_PARTITIONS,
                                          DEFAULT_LAKEFS_CONF_GC_NUM_ADDRESS_PARTITIONS
                                         )
-
     val expiredAddresses = gc
       .getExpiredAddresses(repo,
-                           storageNS,
-                           runID,
-                           gcCommitsLocation,
-                           numRangePartitions,
-                           numAddressPartitions
-                          )
+        storageNS,
+        runID,
+        gcCommitsLocation,
+        numRangePartitions,
+        numAddressPartitions
+      )
       .toDF("address")
       .withColumn(MARK_ID_KEY, lit(markID))
+      .cache()
 
     spark.conf.set("spark.sql.sources.partitionOverwriteMode", "dynamic")
     expiredAddresses.write
       .partitionBy(MARK_ID_KEY)
       .mode(SaveMode.Overwrite)
       .parquet(gcAddressesLocation)
 
-    writeAddressesMarkMetadata(runID, markID, gcAddressesLocation, gcCommitsLocation)
-
     println("Expired addresses:")
     expiredAddresses.show()
+
+    // write expired addresses as text - write to the "[addresses path] + .text"
```

Comment:
Can you please document that this is needed for b&r with rclone?

Reply:
Done